agent/unit_state.go

```diff
@@ -71,12 +73,19 @@ type UnitStatePublisher struct {
 // them to the Registry every 5s. Heartbeat objects are also published as they
 // are received on the channel.
 func (p *UnitStatePublisher) Run(beatchan <-chan *unit.UnitStateHeartbeat, stop chan bool) {
+	var publishPeriod time.Duration
+	if p.ttl > 30*time.Second {
```

Comment:
So we are deliberately not running the publisher for 9mins, 50seconds when we have a TTL of 10mins? That doesn't make sense to me. Why don't we use `ttl / 2`? Which would give us `5 mins` to do the publish before the ttl kicks in.